src/client/actions/user/index.ts

```diff
@@ -1,34 +1,64 @@
 import moment from 'moment-timezone'
 
+import querystring from 'querystring'
+/* eslint-disable-next-line import/no-extraneous-dependencies */
+import { History } from 'history'
```

Comment:
how does this import get resolved?

Reply:
It's used so that typescript recognises history.push in this file

Reply:
i was looking at MDN, the history api doesn't have the `push` method that we are using actually... this is mysterious https://developer.mozilla.org/en-US/docs/Web/API/History_API

Reply:
maybe the type we want is actually derived from react-router? [link](https://reacttraining.com/react-router/web/api/history)

Reply:
@liangyuanruo i've discussed with jason regarding this, we've added 'history' into package.json as an explicit dependency so we should be good in terms of module resolution